utils/text/operations.cpp

```diff
@@ -46,25 +49,33 @@ text::escape_xml(const std::string& in)
 {
     std::ostringstream quoted;
 
-    const char* delims = "\"&<>'";  // Keep in sync with 'switch' below.
-    std::string::size_type start_pos = 0;
-    std::string::size_type last_pos = in.find_first_of(delims);
-    while (last_pos != std::string::npos) {
-        quoted << in.substr(start_pos, last_pos - start_pos);
-        switch (in[last_pos]) {
-        case '"':  quoted << "&quot;"; break;
-        case '&':  quoted << "&amp;"; break;
-        case '<':  quoted << "&lt;"; break;
-        case '>':  quoted << "&gt;"; break;
-        case '\'': quoted << "&apos;"; break;
-        default:   UNREACHABLE;
+    for (std::string::const_iterator it = in.begin();
+         it != in.end(); ++it) {
+        unsigned char c = (unsigned char)*it;
+        if (c == '"') { 
+            quoted << "&quot;";
+        } else if (c == '&') { 
+            quoted << "&amp;";
+        } else if (c == '<') { 
+            quoted << "&lt;";
+        } else if (c == '>') { 
+            quoted << "&gt;";
+        } else if (c == '\'') { 
+            quoted << "&apos;";
```

Comment:
I respected those entities, because the old code did that.
Changing this behavior to emit ASCII for these characters would be a change
in behavior, and would require updating the *utils/text/operations_test:escape_xml__some_escaping* unit test.
I didn't want to go down that path, but if you want that, and allow me to change the unit test, I can do that.